lib/improver/spotdata/read_input.py

```diff
@@ -144,14 +142,22 @@ def get_additional_diagnostics(diagnostic_name, diagnostic_data_path,
                       data, with a single entry is time_extract is provided.
 
     """
-    cubes = Load('multi_file').process(
-        diagnostic_data_path + '/*/*' + diagnostic_name + '*',
-        None)
+    # Search directory structure for all files relevant to current diagnostic.
+    files_to_read = [
```

Comment:
I think this may lead to problems down the line. I think the code should be called with the base directory or glob directly rather than recursively delving into something it's given. It's quite easy to mess up environment variables in suites, and we could get hard-to-notice behaviour we don't want with this (e.g. trying to read every file in the share/cycle directory) - there's also the filesystem metadata side to this.